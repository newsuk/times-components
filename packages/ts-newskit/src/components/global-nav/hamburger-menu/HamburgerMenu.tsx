import React, { useState } from 'react';
import { MenuDivider, Block, Visible, TextBlock } from 'newskit';
import NavButtonSection from './NavButtons';
import NavigationList from './NavigationList';
import { HamburgerMenuNav } from '../styles';
import { NavigationData } from '../types';

const HamburgerMenu: React.FC<{
  isLoggedIn?: boolean;
  data: NavigationData;
}> = ({ isLoggedIn, data }) => {
  const mainNavigation = 'Sections';

  const [expandedL1, setExpandedL1] = useState<string>('');
  const [selected, setSelected] = useState(mainNavigation);

  return (
    <HamburgerMenuNav
      aria-label="menu-vertical"
      vertical
      align="spaceBetween"
      overrides={{ spaceInline: 'space000' }}
    >
      <Visible xs sm>
        <NavButtonSection
          loggedIn={isLoggedIn}
          setSelected={setSelected}
          selected={selected}
        />
      </Visible>
      <NavigationList
        data={
          selected === mainNavigation
            ? data.mainMenuItems
            : data.accountMenuItems
        }
        onExpand={setExpandedL1}
        expandedL1={expandedL1}
      />
      {selected === mainNavigation ? (
        <>
          <Block
            stylePreset="blockWrapper"
            paddingInline="space040"
            paddingBlock="space040"
          >
            <TextBlock typographyPreset="newPreset010">More</TextBlock>
          </Block>
          <MenuDivider />
          <NavigationList data={data.moreMenuItems} />
        </>
      ) : null}
    </HamburgerMenuNav>
  );
};

export default HamburgerMenu;
