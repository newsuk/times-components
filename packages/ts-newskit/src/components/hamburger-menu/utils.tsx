import React from 'react';
import { Story } from '@storybook/addon-docs';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

export const StoryWithTheme: React.FC<{ name: string }> = ({
  name,
  children
}) => (
  <ThemeProvider theme={TimesWebLightTheme}>
    <Story name={name}>{children}</Story>
  </ThemeProvider>
);
