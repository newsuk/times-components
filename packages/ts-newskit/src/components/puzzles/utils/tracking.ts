import { PuzzleScrollClickHandlerType, MouseEventType } from '../cards-container/types';

type TrackingPuzzleCategoryProps = {
  nextPrev: string;
  puzzleType: string;
};

export const puzzleCategoryClickTracking = (
  event: MouseEventType,
  puzzleScroll: TrackingPuzzleCategoryProps,
  clickHandler: PuzzleScrollClickHandlerType
) => {
  puzzleScroll && clickHandler(event, puzzleScroll);
};
