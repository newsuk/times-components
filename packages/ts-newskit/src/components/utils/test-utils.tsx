import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={TimesWebLightTheme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
