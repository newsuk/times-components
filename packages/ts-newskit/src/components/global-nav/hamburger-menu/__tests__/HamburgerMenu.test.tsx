import React from 'react';
import { fireEvent } from '@testing-library/react';
import { render, screen } from '../../../utils/test-utils';
import '@testing-library/jest-dom';
import data from '../../__tests__/fixtures/test-data.json';

import { HamburgerMenu } from '../HamburgerMenu';

describe('HamburgerMenu', () => {
  it('should render the component', () => {
    const { asFragment } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should expand the L1 you click on it', () => {
    const { getByText, getAllByTestId } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(getByText('Item 1')).not.toBeVisible();
    const Button = getAllByTestId('menu-sub-button')[0];
    fireEvent.click(Button);
    expect(getByText('Item 1')).toBeVisible();
  });
  it('should close the expanded L1 if you click on it again', () => {
    const { getByText, getAllByTestId } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(getByText('Item 1')).not.toBeVisible();
    const Button = getAllByTestId('menu-sub-button')[0];
    fireEvent.click(Button);
    expect(getByText('Item 1')).toBeVisible();
    fireEvent.click(Button);
    expect(getByText('Item 1')).not.toBeVisible();
  });
  it('should close the expanded L1 if you click on another L1', () => {
    const { getByText, getAllByTestId } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(getByText('Item 1')).not.toBeVisible();
    const Button1 = getAllByTestId('menu-sub-button')[0];
    const Button2 = getAllByTestId('menu-sub-button')[1];
    fireEvent.click(Button1);
    expect(getByText('Item 1')).toBeVisible();
    fireEvent.click(Button2);
    expect(getByText('Item 1')).not.toBeVisible();
  });
});

describe('HamburgerMenu - Logged In', () => {
  it('should render the Sections and account section on smaller devices', () => {
    const { asFragment, getByText } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText('Sections')).toBeVisible();
    expect(getByText('My account')).toBeVisible();
  });

  it('should default to show the Sections', () => {
    const { getByText, queryByText } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(queryByText('Account Menu 1')).toBeFalsy();
    expect(getByText('Main Menu 1')).toBeVisible();
    expect(getByText('More 1')).toBeVisible();
  });

  it('should change the nav items if you click onto the Logged in Menu buttons', () => {
    const { getByText, queryByText } = render(
      <HamburgerMenu data={data} isLoggedIn={true} />
    );
    expect(getByText('Main Menu 1')).toBeVisible();
    expect(queryByText('Account Menu 1')).toBeFalsy();
    const myAccountButton = getByText('My account');
    fireEvent.click(myAccountButton);
    expect(getByText('Account Menu 1')).toBeVisible();
    expect(queryByText('Main Menu 1')).toBeFalsy();
    const SectionsButton = getByText('Sections');
    fireEvent.click(SectionsButton);
    expect(getByText('Main Menu 1')).toBeVisible();
    expect(queryByText('Account Menu 1')).toBeFalsy();
  });
});

describe('HamburgerMenu - Logged Out', () => {
  it('should render the login and subscribe button section on smaller devices', () => {
    const { asFragment, getByText } = render(
      <HamburgerMenu data={data} isLoggedIn={false} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText('Log in')).toBeVisible();
    expect(getByText('Subscribe')).toBeVisible();
  });

  it('should only show the Sections and not the account menu', () => {
    const { getByText, queryByText } = render(
      <HamburgerMenu data={data} isLoggedIn={false} />
    );
    expect(queryByText('Account Menu 1')).toBeFalsy();
    expect(getByText('Main Menu 1')).toBeVisible();
    expect(getByText('More 1')).toBeVisible();
  });
});

describe('Search field', () => {
  it('contains the search bar', () => {
    const { getByPlaceholderText } = render(<HamburgerMenu data={data} />);
    expect(getByPlaceholderText('Search times.co.uk')).toBeVisible();
  });

  it('should update search field value', async () => {
    render(<HamburgerMenu data={data} />);

    const searchField = screen.getByPlaceholderText('Search times.co.uk');

    fireEvent.change(searchField, {
      target: { value: 'Test Value' }
    });
    expect(searchField.getAttribute('value')).toEqual('Test Value');
  });

  it('should clear search field when clicked', async () => {
    render(<HamburgerMenu data={data} />);

    const searchField = screen.getByPlaceholderText('Search times.co.uk');

    fireEvent.change(searchField, {
      target: { value: 'Test Value' }
    });

    const clearSearchBtn = screen.getByRole('button', { name: 'Clear search' });
    expect(searchField).toBeVisible();
    expect(clearSearchBtn).toBeVisible();

    fireEvent.click(clearSearchBtn);

    expect(searchField.getAttribute('value')).toEqual('');
  });
});
