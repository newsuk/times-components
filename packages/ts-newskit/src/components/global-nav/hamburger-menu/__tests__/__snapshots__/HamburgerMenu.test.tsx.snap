// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu - Logged In should render the Sections and account section on smaller devices 1`] = `
<DocumentFragment>
  <nav
    aria-label="menu-vertical"
    class="css-oyjuht"
    data-testid="menu-container"
    role="navigation"
  >
    <ul
      role="list"
    >
      <div
        class="css-gre9re"
      >
        <div
          aria-label="Search Bar"
          class="css-nang7u"
          role="region"
        >
          <div
            class="css-93pna7"
          >
            <div
              class="css-wjzosl"
            >
              <svg
                class="css-192eire"
                fill="none"
                height="16"
                overrides="[object Object]"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15.022 16-5.844-5.844a4.9 4.9 0 0 1-1.556.9 5.5 5.5 0 0 1-1.889.322c-1.6 0-2.955-.556-4.066-1.667C.556 8.6 0 7.26 0 5.69c0-1.57.556-2.911 1.667-4.022C2.777.556 4.126 0 5.71 0c1.57 0 2.908.556 4.011 1.667 1.104 1.11 1.656 2.452 1.656 4.022 0 .637-.104 1.252-.311 1.844a5.64 5.64 0 0 1-.934 1.667L16 15.022l-.978.978Zm-9.31-5.956c1.2 0 2.221-.425 3.066-1.277.844-.852 1.266-1.878 1.266-3.078S9.622 3.463 8.778 2.61C7.933 1.76 6.91 1.333 5.71 1.333c-1.215 0-2.248.426-3.1 1.278-.852.852-1.278 1.878-1.278 3.078s.426 2.226 1.278 3.078c.852.852 1.885 1.277 3.1 1.277Z"
                  fill="currentcolor"
                />
              </svg>
            </div>
            <input
              aria-describedby="searchTimesLabel"
              class="css-pvilfn"
              placeholder="Search times.co.uk"
              type="text"
              value=""
            />
          </div>
          <span
            class="css-13791zy"
            id="searchTimesLabel"
          >
            Search times.co.uk
          </span>
        </div>
        <nav
          aria-label="Navigation Menu"
          class="css-19o37qh"
          data-testid="menu-container"
          role="region"
        >
          <ul
            role="list"
          >
            <button
              class="css-1234wqe"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1bgi23z"
              >
                Sections
              </span>
            </button>
            <button
              class="css-g6ka6m"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1bgi23z"
              >
                My account
              </span>
            </button>
          </ul>
        </nav>
      </div>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/main-menu-1"
          id="vertical-main-menu-1"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-2"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 2
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-2"
              id="vertical-sub-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-3"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 3
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/menu-3-item-2"
              id="vertical-sub-menu-3-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <div
        class="css-1ybfqts"
      >
        <p
          class="css-9id5ai"
        >
          More
        </p>
      </div>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-1"
          id="vertical-more-1"
        >
          <span
            class="css-1bgi23z"
          >
            More 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-2"
          id="vertical-more-2"
        >
          <span
            class="css-1bgi23z"
          >
            More 2
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`HamburgerMenu - Logged Out should render the login and subscribe button section on smaller devices 1`] = `
<DocumentFragment>
  <nav
    aria-label="menu-vertical"
    class="css-oyjuht"
    data-testid="menu-container"
    role="navigation"
  >
    <ul
      role="list"
    >
      <div
        class="css-gre9re"
      >
        <div
          aria-label="Navigation Menu"
          class="css-1q6m5b0"
          role="region"
        >
          <div
            class="css-14f0ndj"
          >
            <button
              class="css-55uab"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1e6qfd8"
              >
                Log in
              </span>
            </button>
            <div
              class="css-b6vv9c"
            />
            <button
              class="css-1yh0br7"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1e6qfd8"
              >
                Subscribe
              </span>
            </button>
          </div>
        </div>
        <div
          aria-label="Search Bar"
          class="css-w3cs1x"
          role="region"
        >
          <div
            class="css-93pna7"
          >
            <div
              class="css-wjzosl"
            >
              <svg
                class="css-192eire"
                fill="none"
                height="16"
                overrides="[object Object]"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15.022 16-5.844-5.844a4.9 4.9 0 0 1-1.556.9 5.5 5.5 0 0 1-1.889.322c-1.6 0-2.955-.556-4.066-1.667C.556 8.6 0 7.26 0 5.69c0-1.57.556-2.911 1.667-4.022C2.777.556 4.126 0 5.71 0c1.57 0 2.908.556 4.011 1.667 1.104 1.11 1.656 2.452 1.656 4.022 0 .637-.104 1.252-.311 1.844a5.64 5.64 0 0 1-.934 1.667L16 15.022l-.978.978Zm-9.31-5.956c1.2 0 2.221-.425 3.066-1.277.844-.852 1.266-1.878 1.266-3.078S9.622 3.463 8.778 2.61C7.933 1.76 6.91 1.333 5.71 1.333c-1.215 0-2.248.426-3.1 1.278-.852.852-1.278 1.878-1.278 3.078s.426 2.226 1.278 3.078c.852.852 1.885 1.277 3.1 1.277Z"
                  fill="currentcolor"
                />
              </svg>
            </div>
            <input
              aria-describedby="searchTimesLabel"
              class="css-pvilfn"
              placeholder="Search times.co.uk"
              type="text"
              value=""
            />
          </div>
          <span
            class="css-13791zy"
            id="searchTimesLabel"
          >
            Search times.co.uk
          </span>
        </div>
      </div>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/main-menu-1"
          id="vertical-main-menu-1"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-2"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 2
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-2"
              id="vertical-sub-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-3"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 3
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/menu-3-item-2"
              id="vertical-sub-menu-3-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <div
        class="css-1ybfqts"
      >
        <p
          class="css-9id5ai"
        >
          More
        </p>
      </div>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-1"
          id="vertical-more-1"
        >
          <span
            class="css-1bgi23z"
          >
            More 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-2"
          id="vertical-more-2"
        >
          <span
            class="css-1bgi23z"
          >
            More 2
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`HamburgerMenu should render the component 1`] = `
<DocumentFragment>
  <nav
    aria-label="menu-vertical"
    class="css-oyjuht"
    data-testid="menu-container"
    role="navigation"
  >
    <ul
      role="list"
    >
      <div
        class="css-gre9re"
      >
        <div
          aria-label="Search Bar"
          class="css-nang7u"
          role="region"
        >
          <div
            class="css-93pna7"
          >
            <div
              class="css-wjzosl"
            >
              <svg
                class="css-192eire"
                fill="none"
                height="16"
                overrides="[object Object]"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15.022 16-5.844-5.844a4.9 4.9 0 0 1-1.556.9 5.5 5.5 0 0 1-1.889.322c-1.6 0-2.955-.556-4.066-1.667C.556 8.6 0 7.26 0 5.69c0-1.57.556-2.911 1.667-4.022C2.777.556 4.126 0 5.71 0c1.57 0 2.908.556 4.011 1.667 1.104 1.11 1.656 2.452 1.656 4.022 0 .637-.104 1.252-.311 1.844a5.64 5.64 0 0 1-.934 1.667L16 15.022l-.978.978Zm-9.31-5.956c1.2 0 2.221-.425 3.066-1.277.844-.852 1.266-1.878 1.266-3.078S9.622 3.463 8.778 2.61C7.933 1.76 6.91 1.333 5.71 1.333c-1.215 0-2.248.426-3.1 1.278-.852.852-1.278 1.878-1.278 3.078s.426 2.226 1.278 3.078c.852.852 1.885 1.277 3.1 1.277Z"
                  fill="currentcolor"
                />
              </svg>
            </div>
            <input
              aria-describedby="searchTimesLabel"
              class="css-pvilfn"
              placeholder="Search times.co.uk"
              type="text"
              value=""
            />
          </div>
          <span
            class="css-13791zy"
            id="searchTimesLabel"
          >
            Search times.co.uk
          </span>
        </div>
        <nav
          aria-label="Navigation Menu"
          class="css-19o37qh"
          data-testid="menu-container"
          role="region"
        >
          <ul
            role="list"
          >
            <button
              class="css-1234wqe"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1bgi23z"
              >
                Sections
              </span>
            </button>
            <button
              class="css-g6ka6m"
              data-testid="button"
              type="button"
            >
              <span
                class="css-1bgi23z"
              >
                My account
              </span>
            </button>
          </ul>
        </nav>
      </div>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/main-menu-1"
          id="vertical-main-menu-1"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-2"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 2
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-2"
              id="vertical-sub-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <button
          aria-expanded="false"
          class="css-17yq0k1"
          data-testid="menu-sub-button"
          id="vertical-main-menu-3"
          type="button"
        >
          <span
            class="css-1bgi23z"
          >
            Main Menu 3
          </span>
          <svg
            aria-hidden="true"
            class="css-bl48kn-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
        <ul
          class="css-n5iuvv"
        >
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/item-1"
              id="vertical-sub-item-1"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 1
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
          <li
            class="nk-menu-item css-1gz71w1"
          >
            <a
              class="css-1xcetor"
              data-testid="buttonLink"
              href="/menu-3-item-2"
              id="vertical-sub-menu-3-item-2"
            >
              <span
                class="css-1bgi23z"
              >
                Menu 3 Item 2
              </span>
            </a>
          </li>
          <li
            aria-hidden="true"
            class="css-2zdhyu"
            role="separator"
          >
            <hr
              aria-hidden="true"
              class="css-1s3b7bb"
              data-testid="divider"
            />
          </li>
        </ul>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <div
        class="css-1ybfqts"
      >
        <p
          class="css-9id5ai"
        >
          More
        </p>
      </div>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-1"
          id="vertical-more-1"
        >
          <span
            class="css-1bgi23z"
          >
            More 1
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
      <li
        class="nk-menu-item css-1gz71w1"
      >
        <a
          class="css-10u3zx8"
          data-testid="buttonLink"
          href="/more-2"
          id="vertical-more-2"
        >
          <span
            class="css-1bgi23z"
          >
            More 2
          </span>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="css-2zdhyu"
        role="separator"
      >
        <hr
          aria-hidden="true"
          class="css-1s3b7bb"
          data-testid="divider"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
